[css-view-transitions-2] root-element-transition-iframe-with-startVT-on-main.html fails.

mattwoodrow opened the pull request:
#### 5f7b390a6aa8206c7af46cbd61d5afc9912b3374
<pre>
[css-view-transitions-2] root-element-transition-iframe-with-startVT-on-main.html fails.
<a href="https://bugs.webkit.org/show_bug.cgi?id=279160">https://bugs.webkit.org/show_bug.cgi?id=279160</a>
&lt;<a href="https://rdar.apple.com/135307657">rdar://135307657</a>&gt;

Reviewed by Tim Nguyen.

The test currently has the default 0.5 second animation on the addition of the
border of the iframe (which is a size change, so the group is animating the
width/height values).

The test waits for the inner doc to navigate, but not for the outer animation,
so it racily captures in the middle of the animation.

Set the animation duration to 0s so that it completes immediately (but we remain
in a ViewTransition, since the duration on the root element&apos;s pseudo group is
still 300s).

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/navigation/root-element-transition-iframe-with-startVT-on-main.html:

Canonical link: <a href="https://commits.webkit.org/283196@main">https://commits.webkit.org/283196@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cbb9d46e997f2cb2870910fb90e6e14fbc76958d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/65499 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/44870 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/18117 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/69525 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/16108 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/52671 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/16388 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/52594 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/11166 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/68566 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/41450 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/56696 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/33218 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/38128 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/14074 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/14984 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/59961 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/14416 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/71230 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/9453 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/13859 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/59911 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/9485 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/56761 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/60183 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/uri-scheme (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14444 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/7808 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/1467 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/40680 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/41756 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/42939 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/41500 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/navigation/root-element-transition-iframe-with-startVT-on-main.html:
```diff
@@ -29,6 +29,9 @@
     opacity: 1;
   }
+  ::view-transition-group(inner) {
+    animation-duration: 0s;
```
`animation-play-state: paused` ?

mattwoodrow replied:
I think we'd need to set it to the end too?

nt1m replied:
Oh I guess maybe the play-state wouldn't inherit. Whatever works and isn't flaky seems fine to me.

mattwoodrow committed:
[css-view-transitions-2] root-element-transition-iframe-with-startVT-on-main.html fails.
https://bugs.webkit.org/show_bug.cgi?id=279160
<rdar://135307657>

Reviewed by Tim Nguyen.

The test currently has the default 0.5 second animation on the addition of the
border of the iframe (which is a size change, so the group is animating the
width/height values).

The test waits for the inner doc to navigate, but not for the outer animation,
so it racily captures in the middle of the animation.

Set the animation duration to 0s so that it completes immediately (but we remain
in a ViewTransition, since the duration on the root element's pseudo group is
still 300s).

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-view-transitions/navigation/root-element-transition-iframe-with-startVT-on-main.html:

Canonical link: https://commits.webkit.org/283196@main
